test/whitelist_generator_test.py

```diff
@@ -0,0 +1,58 @@
+import pytest
+from covidactnow.datapublic.common_test_helpers import to_dict
+from libs.datasets import combined_datasets
+from libs.datasets.timeseries import TimeseriesDataset
+from pyseir.inference.whitelist_generator import WhitelistGenerator
+from test.dataset_utils_test import read_csv_and_index_fips_date
+
+# turns all warnings into errors for this module
+pytestmark = pytest.mark.filterwarnings("error")
+
+
+@pytest.mark.slow
+def test_all_data_smoke_test():
+    input_timeseries = combined_datasets.load_us_timeseries_dataset().get_subset(state="AL")
+    df = WhitelistGenerator().generate_whitelist(input_timeseries)
+    assert not df.empty
+
+
+def test_skip_gaps_in_cases_and_deaths_metrics():
+    input_df = read_csv_and_index_fips_date(
+        "fips,country,state,county,aggregate_level,date,cases,deaths\n"
+        "97111,US,ZZ,Bar County,county,2020-04-01,10,1\n"
+        "97111,US,ZZ,Bar County,county,2020-04-02,,2\n"
+        "97111,US,ZZ,Bar County,county,2020-04-03,30,\n"
+        "97111,US,ZZ,Bar County,county,2020-04-04,40,4\n"
+    )
+
+    df = WhitelistGenerator().generate_whitelist(TimeseriesDataset(input_df.reset_index()))
+
+    assert to_dict(["fips"], df) == {
+        "97111": {"state": "ZZ", "county": "Bar County", "inference_ok": False},
+    }
+
+
+def test_inference_ok_with_5_days_cases_changed():
+    # 5 days with cases data isn't enough to make inference_ok, 6 days are
+    # needed so that there are 5 days with an *delta* relative to a previous day.
+    input_df = read_csv_and_index_fips_date(
```

Comment:
any reason not to just use common_df.read_csv?

Reply:
Not very strong ones... so the test doesn't need to call StringIO and consistency with read_csv_and_index_fips that doesn't parse dates.